### ISSUE TITLE
Does not produce CSS

### ISSUE THREAD

**Issue by edwardhotchkiss** [christophercliff/metalsmith-less#19]
The plugin does not produce any CSS file. Here is my usage:

```javascript
.use(less({
    pattern: [
      'bower_components/bootstrap/less/bootstrap.less',
      'src/less/app.less'
    ],
    render: {
      paths: [
        'bower_components/bootstrap/less/**/*.less',
      ],
    }
  }))
```

Any ideas?

**Comment by christophercliff** [comment]
See https://github.com/christophercliff/metalsmith-less/blob/0c858b41a4eec3a30785434c30b9b08458130ff0/test/index.js#L17-L32 for example `paths` usage.

**Comment by edwardhotchkiss** [comment]
Following your pattern I have no change. What would you change to make this work?

**Comment by jmjf** [comment]
I offer this to save other users a lot of time, hopefully.

After hacking around for a day trying to figure out how this beast works, the following output bootstrap.css in build/css

```
var ... // all the requires
    less = require('metalsmith-less'),
    ignore = require('metalsmith-ignore');
    ... // the usual build pipeline setup
	.use(less({
		pattern: 'less/bootstrap.less',
		render: {
			paths: [
				'./src/less'
			]
		}
	}))
    ... // rest of the build pipeline
	.use(ignore([
		'**/*.less'
	]))
    // Serve and Watch go here if you're using them
	.build(function (err) {
		if (err) {
			console.log(err);
		}
		else {
			console.log('Site build complete!');
		}
	});

```

with the following directory tree

```
project
    src
        less
            bootstrap.less
    vendor
        bootstrap
            less
                bootstrap.less
```

where bootstrap.less is

```
@import '../../vendor/bootstrap/less/bootstrap.less';
```

The ignore plugin ensures less files aren't copied to build. As I get past tool setup and into development, I'll copy Bootstrap's bootstrap.less into src/less/bootstrap.less, comment the parts I don't need, and add a custom variables.less.

**Conclusions** 
- The `render.paths` option is relative to the project root, not src.
- The `render.paths` option defines the cwd path to which all file references in less files (`@import` statements) are relative and should be considered required to use this plugin sanely.
- The most intuitive approach for imports is to have a `src/less` directory containing all project less files and set `render.paths` to point to `./src/less`. This approach makes all import paths relative to `src/less`, where the less files are.
- The `pattern` option specifies the files to compile to `build/css`.
- The default `**/*.less` pattern will compile all less files in the `src` directory individually.
- As a result, if you are importing any less file in `src` into any other less file in `src`, set `pattern` to a list of specific files to compile to `build/css`.
- The `pattern` option is relative to `src`, not `render.paths`. If you are putting less files in `src/less`, all patterns will be like 'less/lessfile.less'.
- To import a css file, use `@import (inline) "path/to/file.css"`;. (See the [LESS docs](http://lesscss.org/features/#import-directives-feature)).
- The `metalsmith-ignore` plugin is necessary if you aren't using a secondary build tool to clean up the `build` directory.

Some of these points are documented more or less clearly in various issues. It would probably be best if this information was in the README.md to save people time and frustration. I'll set up a PR for it if I get a chance unless someone beats me to it.

**Comment by edwardhotchkiss** [comment]
You're a God. Thank you sir.

**Comment by Sequoia** [comment]
Even with these instructions I can't get it to output any css :(
```
$ tree src
src
├── img
│   └── sequoia.jpg
├── index.md
├── less
│   ├── clean-blog.less
│   ├── mixins.less
│   └── variables.less
└── posts

3 directories, 5 files

$ tree site
site
├── css
│   ├── clean-blog.css
│   ├── mixins.css
│   └── variables.css
├── img
│   └── sequoia.jpg
├── index.html
└── less
    ├── clean-blog.less
    ├── mixins.less
    └── variables.less

3 directories, 8 files

$ head src/less/clean-blog.less
@import "./variables";
@import "./mixins";

// Global Components

body {
	.serif;
	font-size: 20px;
	color: @gray-dark;
}

$ ls -la site/css
...
-rw-r--r--  1 sequoia  staff    0 Aug 17 16:43 clean-blog.css
-rw-r--r--  1 sequoia  staff    0 Aug 17 16:43 mixins.css
-rw-r--r--  1 sequoia  staff    0 Aug 17 16:43 variables.css

```

my build:
```javascript
Metalsmith(__dirname)
    //.use(logFilenames)
    .use(markdown())
    .use(less({
      render: {
        pattern: 'less/clean-blog.less',
        paths: [
            path.join(__dirname, '/src/less')
        ]
      }
    }))
    .use(myLayouts())
    .destination('site')
    .build(handleError);
```

Where am I going wrong? I followed the thread all the way into less.render which is returning an empty string for CSS, presumably some bad configs are being passed in but from where?

**Comment by Sequoia** [comment]
Turns out running layout after this conflicts with less. changed order.

**Comment by jmjf** [comment]
FWIW, I've shifted to a Gulp-centric build process. My gulpfile.js runs Metalsmith to build the HTML as one of the build steps, but everything else is Gulp-based. That's more about specific things I needed to do in the build process (responsive image resizing and optimization, cache busting with rev-stamping) than any particular difficulties with Metalsmith. Gulp's pipeline concept is very similar to Metalsmith's, so there isn't a lot of conceptual translation between the two like I ran into with Grunt.

I looked at gulpsmith and some of the plugins that support using one tool's plugins with the others but didn't like the impedance mismatches I saw described. The gulp plugins to run Metalsmith seem to assume a JSON-based build, which won't work for me (Handlebars helpers, multi-targeting dev/test/prod with a single script), so I just spawn metalsmith with child_process in one of my Gulp tasks.
